src/server/routes/actions.ts

```diff
@@ -26,19 +33,83 @@ router.get<unknown, BalanceResponse>("/balance", (_, res) => {
     });
 });
 
+const missingParameterError = (res: Response<DripErrorResponse>, parameter: string): void => {
+  res.status(400).send({ error: `Missing parameter: '${parameter}'` });
+};
+
+const addressMiddleware = (
+  req: Request<unknown, DripResponse, Partial<DripRequestType>>,
+  res: Response,
+  next: NextFunction,
+): void => {
+  if (!req.body.address) {
+    return missingParameterError(res, "address");
+  }
+  next();
+};
+
 const dripRequestHandler = new DripRequestHandler(actions, storage, recaptchaService);
 
+router.post<unknown, DripResponse, FaucetRequestType>("/faucet", addressMiddleware, async (req, res) => {
```

Comment:
Consider having a Partial here same as the previous endpoint, because no matter what types we put here, we could be fooled by typescript because this input comes from external input.

```suggestion
router.post<unknown, DripResponse, Partial<FaucetRequestType>>("/faucet", addressMiddleware, async (req, res) => {
```

For example right now you're treating `address` as defined, but it actually could be undefined if the external user does not supply it in the API call.

Reply:
The reason for which I'm not checking address is because the `addressMiddleware` is doing so.

I'm not using `partial` as it still gonna throw typescript compilation errors if I do so because it assumes I didn't check that.

I'm still checking all the variables, I'm just verifying them in shared code.

Reply:
Ahh I missed the `addressMiddleware` part yesterday - it makes sense now.